voicebox/__main__.py

```diff
@@ -10,25 +10,48 @@
     format='%(asctime)s %(name)s %(levelname)-8s  %(message)s',
     datefmt='(%H:%M:%S)'
 )
-
-
-def main():
-
-    """
-    Main function to start system
-
-    provides a command line interface for interracting
-    with the voicebox APIs.
-    """
-
+server = Server()
+async def run(port, bootstrap_ip=None, bootstrap_port=None):
+    await server.listen(port)
+    if bootstrap_ip and bootstrap_port:
+        try:
+            await server.bootstrap([(bootstrap_ip, bootstrap_port)])
+        except Exception as e:
+            logging.error(f"Error bootstrapping with node {bootstrap_ip}:{bootstrap_port} - {e}")
+            return False
+    return True
+
+async def setusername(username, ip, port):
+    result = await server.get(username)
+    if result is not None:
+        return False    
+    await server.set(username, ip + ":" + str(port))
+    return True
+    
+async def getusername(username):
```

Comment:
The `getusername` function returns an empty string if the username is not found. It would be more informative to return a more descriptive error message. [medium]